Source/WebCore/css/StyleRule.cpp

```diff
@@ -199,30 +199,48 @@ Ref<CSSRule> StyleRuleBase::createCSSOMWrapper(CSSStyleSheet* parentSheet, CSSGr
 
 unsigned StyleRule::averageSizeInBytes()
 {
-    return sizeof(StyleRule) + sizeof(CSSSelector) + StyleProperties::averageSizeInBytes();
+    return sizeof(StyleRule) + sizeof(CSSSelector) + StyleProperties::averageSizeInBytes() + sizeof(WeakPtr<StyleRule>) + sizeof(Vector<Ref<StyleRuleBase>>);
 }
 
-StyleRule::StyleRule(Ref<StyleProperties>&& properties, bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors)
+StyleRule::StyleRule(Ref<StyleProperties>&& properties, bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors, WeakPtr<StyleRule> parentRule, Vector<Ref<StyleRule>> nestedRules)
     : StyleRuleBase(StyleRuleType::Style, hasDocumentSecurityOrigin)
     , m_properties(WTFMove(properties))
     , m_selectorList(WTFMove(selectors))
+    , m_parentRule(parentRule)
+    , m_nestedRules(nestedRules)
+{
+}
+
+StyleRule::StyleRule(bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors)
+    : StyleRuleBase(StyleRuleType::Style, hasDocumentSecurityOrigin)
+    , m_properties(MutableStyleProperties::createEmpty())
+    , m_selectorList(WTFMove(selectors))
 {
 }
 
 StyleRule::StyleRule(const StyleRule& o)
     : StyleRuleBase(o)
     , m_properties(o.properties().mutableCopy())
     , m_selectorList(o.m_selectorList)
+    , m_parentRule(o.m_parentRule)
+    , m_nestedRules(o.m_nestedRules)
     , m_isSplitRule(o.m_isSplitRule)
     , m_isLastRuleInSplitRule(o.m_isLastRuleInSplitRule)
 {
 }
 
 StyleRule::~StyleRule() = default;
 
-Ref<StyleRule> StyleRule::create(Ref<StyleProperties>&& properties, bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors)
+void StyleRule::setProperties(Ref<StyleProperties> properties) { m_properties = properties; }
```

Comment:
We don't usually define methods in implementation files in one line like this. Could this be in the header file instead? Otherwise we should split it into three lines so it's easier to debug.

Reply:
Yes the goal was to avoid including `StyleProperties.h` in the header but indeed it's not following our guidelines. Fixed.